Decor/Call.cs

```diff
@@ -80,6 +80,11 @@ public async Task Next()
                             await task.ConfigureAwait(false);
                         }
 
+                        if (task.IsFaulted && task.Exception != null)
+                        {
+                            throw task.Exception;
```

Comment:
This will lose original stacktrace, there is an extension method for rethrowing excpetions without modifying stacktrace, could you change this to `task.Exception.Rethrow()`?